### PR TITLE
Make TimeParameterization classes polymorphic

### PR DESCRIPTION
**Pull request by JafarAbdi**
### Description

Cherry-pick https://github.com/ros-planning/moveit/pull/3021 in moveit2

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Review by stephanie-eng — APPROVED** [review]

**Review by AndyZe — APPROVED** [review]

File: moveit_core/trajectory_processing/include/moveit/trajectory_processing/time_parameterization.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#pragma once
```
Comment (henningkayser) [inline moveit_core/trajectory_processing/include/moveit/trajectory_processing/time_parameterization.h:1]: license header?

**Review by henningkayser — APPROVED** [review]
Is this too urgent so that we can't wait for this to get synced? Good change, imo

**Review by jliukkonen — CHANGES_REQUESTED** [review]
Not sure if I can actually block merging on my own but I'm marking this as "request changes" until there are comments. Otherwise it's too easy to forget them :)

**Comment by JafarAbdi** [comment]
Replace with https://github.com/ros-planning/moveit2/pull/1151

**Pull request closed without merging**